content/blog/2025/08/31/2025-08-31-aarav-mahajan-gsoc-gradle-convention-plugin-for-jenkins-plugin-development.adoc

```diff
@@ -0,0 +1,154 @@
+---
+layout: post
+title: "Introducing the Gradle Convention Plugin for Jenkins Plugin Development"
+tags:
+- jenkins
+- gradle
+- kotlin
+- dsl
+- plugin-development
+- build-tools
+- java
+- groovy
+- developer-tools
+- convention-plugin
+
+author: aaravmahajanofficial
+description: "As part of GSoC 2025, discover how the Gradle Convention Plugin makes Jenkins plugin development easier, faster, and more consistent."
+opengraph:
+  image: /images/post-images/2025/08/aaravmahajan/gradle-convention-plugin.png
+---
+
+**Building Jenkins plugins with Gradle just got easier!** + 
+Meet the Jenkins Gradle Convention Plugin → modern, Kotlin-first, convention-based. Plugin development made effortless 🏄‍♂️🏎️. + 
+🌀 Try it today on the link:https://plugins.gradle.org/plugin/io.github.aaravmahajanofficial.jenkins-gradle-convention-plugin[Gradle Plugin Portal]
+
+== Today’s Jenkins Plugin Development Landscape
+
+Apache Maven remains the primary and officially recommended build tool for Jenkins plugin development, powering the vast majority of plugins in the ecosystem. Its strong integration with the Jenkins plugin parent POM, extensive documentation, and mature lifecycle support make it the safest and default choice for most developers.
+
+Gradle, while technically capable and widely adopted across the broader Java ecosystem, has seen limited adoption in the Jenkins community. It offers powerful advantages, including:
+
+* Extensible build logic defined in code rather than rigid XML  
+* Faster build times with incremental builds, caching, and the Gradle Daemon  
+* A developer-friendly experience with Kotlin and Groovy DSLs, plus interactive tooling 
```

Comment:
```suggestion
* Extensible build logic defined in code rather than rigid XML.
* Faster build times with incremental builds, caching, and the Gradle Daemon.
* A developer-friendly experience with Kotlin and Groovy DSLs, plus interactive tooling.
```
Suggestion to add full stops for these lines.